### ISSUE TITLE
`Unshelve` vs `unshelve`.

### ISSUE THREAD

**Issue by ejgallego** [coq/coq#6292]
So this is a general question / tracker about the duplication of some state between the structure `Proof_global.pstate`, `Proof.proof` and the monad `Proofview.proofview` itself. A good example is `Unshelve` vs `unshelve`.

@aspiwack hinted than indeed we should move towards unifying some of this state, but I am afraid I am not knowleable about what the next steps should be.

**Comment by maximedenes** [comment]
@ppedrot @spiwack Can you summarize the current state of proof states?

I heard a rumor saying that there were two notions of shelf, but never looked at it myself. I'd be interested in knowing more.

**Comment by ejgallego** [comment]
:clap: :clap: :clap: :clap:
